src/site/pages/digital/src/utils/DigitalPaste.ts

```diff
@@ -59,16 +64,41 @@ function TransferNewICData(objs: IOObject[], designer: DigitalCircuitDesigner):
     return action;
 }
 
+/**
+ * Maps positions of pasted components based around intially selected item
+ * @param comps Components in clipboard
+ * @param info Circuit info from which to pull camera and input
+ * @returns Target positions for pasting at mouse pos
+ */
+function shiftAllObj(comps: Component[], info: DigitalCircuitInfo): Vector[] {
+    const {camera, input} = info;
+    let worldMousePos = camera.getWorldPos(input.getMousePos());
+    let shift = comps[0].getPos().sub(worldMousePos);
+    return comps.map(o => o.getPos().sub(shift))
```

Comment:
nit: in my head it makes more sense to have `shift = worldMousePos.sub(comps[0].getPos())` and then do `o.getPos().add(shift)`